### ISSUE TITLE
IonicModalNavService does not pass data to modal

### ISSUE THREAD

**Issue by jhlee8804** [trazek/ionic-modal-nav-example#5]
Hi and sorry to my english.
I want to pass data to modal with $stateParams, but IonicModalNavService does not passed.
Here is my code:

``` javascript
// 1. register a route to $stateProvider
$stateProvider.state('comment-list', {
  views: {
    'ionic-modal-nav@': {
      templateUrl: 'views/board/article-comment.html',
      controller: 'MyController as myCtrl'
    }
  },
  param: {
    url: null,
    articleID: null
  }
});

// 2. show and pass data by IonicModalNavService 
IonicModalNavService.show('comment-list', {
    url: vm.store.url,
    articleID: vm.article.id,
  });

// 3. check passed data at MyController, but it is empty.
function MyController($scope, $stateParams) {
  console.log($stateParams.url);  // undefined
  console.log($stateParams.articleID);  // undefined
  ...
}
```

so i change few code:

``` javascript
  key: "show",
  value: function show(modalState, data) {
    ...

    // fire event with passed data
    this._$rootScope.$emit(SHOW_MODAL, data);

    ...
```

``` javascript
function IonicModalNavService($ionicModal, $rootScope, $state, $ionicHistory, $ionicViewSwitcher, modalOptions) {
  ...

  $rootScope.$on(SHOW_MODAL, function (event, data) {
    // set passed data to modal object
    _this._modal.params = data;
    _this._modal.show();
  });

  ...
}
```

so i got passed data at MyController:

``` javascript
function MyController($scope, $stateParams) {

  $scope.$on('modal.shown', function (e, data) {
    console.log(data.params.url);  // http://abc.com
    console.log(data.params.articleID);  // 100
  });

  ...
}
```

Please check this issue. Thanks.

**Comment by trazek** [comment]
The code example you pasted has `param` in the state definition. Try making it `params`:

```
$stateProvider.state('comment-list', {
  views: {
    'ionic-modal-nav@': {
      templateUrl: 'views/board/article-comment.html',
      controller: 'MyController as myCtrl'
    }
  },
  params: {
    url: null,
    articleID: null
  }
});
```

This was addressed [in issue #2 ](https://github.com/trazek/ionic-modal-nav-example/issues/2)

Let me know if you are still having problems

**Comment by jhlee8804** [comment]
Thanks!! That's my mistake :(
It works fine!
